### PR TITLE
Neworder input

### PR DESCRIPTION
**Pull request by lapinlapin**
#### What's this PR do?
Added blocks of input
#### Where should the reviewer start?
views/newOrder/NewOrderInput
#### How should this be manually tested?
https://localhost:8080/
#### What are the relevant tickets?
#237 
#### Screenshots (if appropriate)
image

#### URL
Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/branch-neworder-input

### CODE REVIEW

**Comment by lapinlapin** [comment]
@roma219 I didn’t use simple input, it doesn’t fit the design much, did customization instead

File: src/components/SimpleInput/SimpleInput.vue [hunk 1]
```diff
@@ -3,7 +3,8 @@
     :class="{
       'input--has-icon': icon,
       'input--has-error': error,
-      'input--disabled': disabled
+      'input--disabled': disabled,
+      'input--value': valueMode
```
Comment (roma219) [inline src/components/SimpleInput/SimpleInput.vue]: input--value-mode

File: src/components/SimpleInput/SimpleInput.vue [hunk 2]
```diff
@@ -116,6 +118,14 @@ export default {
     error: {
       type: String,
       default: ''
+    },
+    valueMode: {
+      type: Boolean,
+      default: false
+    },
+    note: {
```
Comment (roma219) [inline src/components/SimpleInput/SimpleInput.vue]: what is this?

File: src/components/SimpleInput/SimpleInput.vue [hunk 3]
```diff
@@ -162,6 +172,9 @@ export default {
     },
     handleBlur() {
       this.$emit('blur', this.$refs.input.value)
+      if (this.valueMode && !this.value) {
```
Comment (roma219) [inline src/components/SimpleInput/SimpleInput.vue]: why are you emitting placeholder?

File: src/components/SimpleInput/SimpleInput.vue [hunk 4]
```diff
@@ -177,6 +190,7 @@ export default {
 
 .input {
   position: relative;
+  text-align: center;
```
Comment (roma219) [inline src/components/SimpleInput/SimpleInput.vue]: change text-align to a separate prop - `:centered="true"`. It's gonna be needed later without value-mode.

File: src/components/SimpleInput/SimpleInput.vue [hunk 5]
```diff
@@ -244,7 +258,6 @@ export default {
   left: 0;
   transition: all 0.3s;
   pointer-events: none;
-  text-transform: uppercase;
```
Comment (roma219) [inline src/components/SimpleInput/SimpleInput.vue]: why?

File: src/components/SimpleInput/SimpleInput.vue [hunk 6]
```diff
@@ -268,4 +281,15 @@ export default {
 .input__tip {
   color: config('colors.text-primary');
 }
+.input--value {
+  padding: 0;
+
+  .input__input {
+    padding-right: 0;
+    text-align: center;
```
Comment (roma219) [inline src/components/SimpleInput/SimpleInput.vue]: why 2 text-aligns?

File: src/views/NewOrder/NewOrderInput.vue [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    },
+    placeholder: {
+      type: String,
+      default: 'BTC'
+    },
+    note: {
+      type: String,
+      required: true
+    },
+    aside: {
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: Again - no component should position itself like that. This component is gonna be used in a lot of situations without need to have magin-left: 1rem. Please remove and position these component by their container in NewOrder.vue.

File: src/views/NewOrder/NewOrderInput.vue [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      type: String,
+      default: 'left'
+    }
+  },
+  data() {
+    return {
+      value: this.placeholder
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: what is this?

Comment (roma219) [reply src/views/NewOrder/NewOrderInput.vue]: remove

File: src/views/NewOrder/NewOrderInput.vue [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    }
+  }
+}
+</script>
+<style lang="scss">
+  .new-order-input {
+    text-align: center;
+    width: 4.375rem;
+    font-size: config('textSizes.xl');
+    margin-top: -0.875rem;
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: remove - see positioning comment

File: src/views/NewOrder/NewOrderInput.vue [hunk 10]
```diff
@@ -0,0 +1,91 @@
+<template>
+  <div
+    :class="{
+      'new-order-input--left': aside === 'left',
+      'new-order-input--right': aside === 'right'
+    }"
+    class="new-order-input"
+  >
+    <div class="new-order-input--title">{{ title }}</div>
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: it is not a modifier (`--title`)

File: src/views/NewOrder/NewOrderInput.vue [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    <SimpleInput
+      v-model="value"
+      :value-mode="true"
+      :title="title"
+      :note="note"
+      :placeholder="placeholder"
+      type="number"
+    />
+    <div class="new-order-input--note">{{ note }}</div>
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: it is not a modifier (`--note`)

File: src/views/NewOrder/NewOrderInput.vue [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+      'new-order-input--right': aside === 'right'
+    }"
+    class="new-order-input"
+  >
+    <div class="new-order-input--title">{{ title }}</div>
+    <SimpleInput
+      v-model="value"
+      :value-mode="true"
+      :title="title"
+      :note="note"
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: what is note?

File: src/views/NewOrder/NewOrderInput.vue [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+    &--left {
+      margin-left: 0.9375rem;
+    }
+
+    &--right {
+      margin-right: 0.9375rem;
+    }
+
+    &--field {
```
Comment (roma219) [inline src/views/NewOrder/NewOrderInput.vue]: not used

File: src/views/NewOrder/NewOrderPercentSelectorItem.vue [hunk 14]
```diff
@@ -0,0 +1,41 @@
+<template>
```
Comment (roma219) [inline src/views/NewOrder/NewOrderPercentSelectorItem.vue]: what is this? This issue has nothing to do with percent selector. And for PercentSelector - separate item component is not needed.

**Review by roma219 — CHANGES_REQUESTED** [review]
- Misses issue number
image
- All PRs should be branched out and target `dev` branch
- Needs some changes - see comments

**Review by roma219 — APPROVED** [review]

**Comment by roma219** [comment]
@lapinlapin there are merge conflicts

**Comment by lapinlapin** [comment]
@roma219

**Pull request merged**